Lib/asyncio/tasks.py

```diff
@@ -931,18 +931,21 @@ def callback():
 
 
 def create_eager_task_factory(custom_task_constructor):
-
-    if "eager_start" not in inspect.signature(custom_task_constructor).parameters:
-        raise TypeError(
-            "Provided constructor does not support eager task execution")
+    """
+    Creates a function suitable for use as a task factory on an event-loop.
+    The tasks created will be started immediately (rather than being first
+    scheduled to an event loop). The constructor argument can be any callable
+    that constructs a Task-compatible object and supports the eager_start
+    keyword argument.
```

Comment:
The callable has to have a signature that is generally compatible with the `Task` constructor signature, right? Should probably say that here.

```suggestion
    scheduled to an event loop). The constructor argument can be any callable
    with signature compatible with ``Task.__init__``, including the ``eager_start``
    keyword argument.
```

Reply:
```suggestion
    scheduled to an event loop). The constructor argument can be any callable
    that returns a ``Task`` compatible object and has a signature compatible
    with ``Task.__init__``, including the ``eager_start`` keyword argument.
```